src/main/java/com/bisnode/opa/TestRegoTask.java

```diff
@@ -1,35 +1,111 @@
 package com.bisnode.opa;
 
 import com.bisnode.opa.configuration.OpaPluginConvention;
+import com.bisnode.opa.junit.OpaToJunitConverter;
 import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.tasks.InputDirectory;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.tooling.TestExecutionException;
+import org.w3c.dom.Document;
 
+import javax.annotation.Nullable;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class TestRegoTask extends DefaultTask {
 
+    public TestRegoTask() {
+        setGroup("opa");
+        setDescription("Run OPA tests in testDir of any policies provided in srcDir");
+    }
+
+    @Nullable
+    private String srcDir;
+    @Nullable
+    private String testDir;
+
     @TaskAction
     public void testRego() {
-        OpaPluginConvention convention = getProject().getConvention().getPlugin(OpaPluginConvention.class);
+        Project project = getProject();
+        Logger logger = getLogger();
+        OpaPluginConvention convention = project.getConvention().getPlugin(OpaPluginConvention.class);
+
+        String src = Optional.ofNullable(srcDir).orElse(convention.getSrcDir());
+        String testSrc = Optional.ofNullable(testDir).orElse(convention.getTestDir());
 
         String location = convention.getLocation();
-        List<String> command = Arrays.asList(location, "test", convention.getSrcDir(), convention.getTestDir());
+        List<String> command = Arrays.asList(location, "test", "--format=json", src, testSrc);
+
+        Process process;
+        try {
+            logger.debug("Running command {}", String.join(" ", command));
+            process = new ProcessBuilder()
+                    .directory(project.getRootDir())
+                    .command(command)
+                    .start();
+
+            int exitCode = process.waitFor();
+
+            try (BufferedReader reader = bufferedInputReader(process.getInputStream())) {
+                String output = reader.lines().collect(Collectors.joining("\n"));
+                Document document = OpaToJunitConverter.fromOpaTestJson(output);
```

Comment:
Design suggestion: `OpaToJunitConverter` could be a normal class with non-static methods, let's call it `OpaTestResult`. It could be created like `OpaTestResult.fromJson(String)`, then later in line 72 it could be used as `opaTestResult.toJUnitReport()` which would return `OpaJUnitReport`, so you can invoke `opaJUnitReport.writeTo(OutputStream)`. That's just a suggestion :)

Reply:
This way you separate and encapsulate behaviors without having to deal with them here.

Reply:
This is solid advice, and I think we could benefit from breaking the OPA test JSON -> JUnit XML into it's own little package at some point, sort of like the [Python script](https://github.com/open-policy-agent/contrib/tree/master/junit) I wrote for that purpose before. That would make this usable in any context and not just for Gradle. I'll create a task for doing that but will leave it here for now.